src/tsp.py

```diff
@@ -31,14 +45,67 @@ def cost(self,index):
              self.dataset["x"][index[-1]], self.dataset["y"][index[-1]])
         return cost
 
-    def solve(self):
+    def solve(self,paths,epoches):
```

Comment:
There should be a PSOSolver class and a TSPProblem class.
Information -> TSPPRoblem containing the data related to the problem, cost function, etc.
TSPSolver -> PSOSolver containing the data and functions related to solving a specific problem